src/app/_validators/sitting-records-form-validator.spec.ts

```diff
@@ -0,0 +1,134 @@
+            control.patchValue({ dateDay: '', dateMonth: '02', dateYear: '2023' });
+            control.get('dateDay')?.markAsTouched();
+            expect(CustomValidators.validateDateFormat(control)).toEqual({ 'date_invalid': true });
+
+            control.patchValue({ dateDay: '01', dateMonth: '', dateYear: '2023' });
+            control.get('dateMonth')?.markAsTouched();
+            expect(CustomValidators.validateDateFormat(control)).toEqual({ 'date_invalid': true });
+
+            control.patchValue({ dateDay: '01', dateMonth: '02', dateYear: '23' });
+            control.get('dateYear')?.markAsTouched();
+            expect(CustomValidators.validateDateFormat(control)).toEqual({ 'date_invalid': true });
+
+        });
+
+        it('should return date_out_of_Valid error when date is after today', () => {
+            control.patchValue({ dateDay: '33', dateMonth: '02', dateYear: '2023' });
+            control.get('dateDay')?.markAsDirty();
+            control.get('dateMonth')?.markAsDirty();
+            control.get('dateYear')?.markAsDirty();
+            expect(CustomValidators.validateDateFormat(control)).toEqual({ 'date_invalid': true });
+
+            control.patchValue({ dateDay: '1', dateMonth: '15', dateYear: '2023' });
+            control.get('dateDay')?.markAsDirty();
+            control.get('dateMonth')?.markAsDirty();
+            control.get('dateYear')?.markAsDirty();
+            expect(CustomValidators.validateDateFormat(control)).toEqual({ 'date_invalid': true });
+
+            control.patchValue({ dateDay: '1', dateMonth: '11', dateYear: '202a' });
+            control.get('dateDay')?.markAsDirty();
+            control.get('dateMonth')?.markAsDirty();
+            control.get('dateYear')?.markAsDirty();
+            expect(CustomValidators.validateDateFormat(control)).toEqual({ 'date_invalid': true });
+        })
+
+        it('should return date_after_today error when date is after today', () => {
+            const today = new Date()
+            control.patchValue({ dateDay: (today.getDate()+1).toString(), dateMonth: (today.getMonth()+1).toString(), dateYear: today.getFullYear().toString()})
+            expect(CustomValidators.validateDateFormat(control)).toEqual({ 'date_after_today': true })
+        })
+
+        it('should return date_after_today error when date is after today', () => {
+            control.patchValue({ dateDay: '01', dateMonth: '02', dateYear: '2020'})
+            expect(CustomValidators.validateDateFormat(control)).toEqual({ 'date_out_of_support': true })
+        })
+
+        it('should return null when date is valid', () => {
+            const today = new Date()
+            control.patchValue({ dateDay: today.getDate().toString(), dateMonth: (today.getMonth()+1).toString(), dateYear: today.getFullYear().toString()})
+            expect(CustomValidators.validateDateFormat(control)).toEqual(null)
+        })
+
+    });
+
+    describe('validateDay', () => {
```

Comment:
Can we remove validateDay, validateMonth and validateYear test as we have removed it from the main file

Reply:
functions removed and added to checks in single function